File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        for(int i=0; i<arr.length; i++){
            arr[i]=greatestFromRight(arr,i+1);
        }
        return arr;
    }
    public int greatestFromRight(int[] arr, int start){
        if(start>=arr.length) return -1;
        int maxElem = arr[start];
        for(int i=start; i<arr.length; i++){
            maxElem = maxElem > arr[i] ? maxElem : arr[i];
        } 
        return maxElem;
    }
}

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        higher=-1
        count=0
        for i in range(len(arr)-1,-1,-1):
            if higher<arr[i]:
                higher,arr[i]=arr[i],higher
            else:
                arr[i]=higher
        return arr
            